src/utils/backendAuthMixin.js

```diff
@@ -0,0 +1,47 @@
+import { mapState } from 'vuex';
+import Backend from './backend';
+import { createDeepLinkUrl } from './util';
+import { client } from './aeternity';
+
+export default (ignoreCallInQuery) => ({
+  computed: mapState(['useSdkWallet', 'account']),
+  ...!ignoreCallInQuery && {
+    async beforeRouteEnter(to, from, next) {
+      const {
+        method, address, challenge, signature,
+      } = to.query;
+      if (!method || !address || !challenge || !signature) {
+        next();
+        return;
+      }
+      let succeed = false;
+      try {
+        await Backend[method](address, { challenge, signature });
+        succeed = true;
+      } finally {
+        next((vm) => {
+          if (succeed && vm.handleBackendSucceedCall) vm.handleBackendSucceedCall(method);
+          vm.$router.replace(window.location.pathname);
+        });
+      }
+    },
+  },
+  methods: {
+    async backendAuth(method, arg, to) {
+      const { challenge } = await Backend[method](this.account, arg);
```

Comment:
I have extracted the first call of the backend to here (it was duplicated before). Also, seems reasonable to update the backend methods signature to always accept the address as a first argument.